### PR TITLE
Basic flavor support

### PR DESCRIPTION
**Pull request by itamaro**
When flavor is specified, all yabtwork workspaces will be under a flavor-sub-dir.

### CODE REVIEW

File: tests/cpp/YSettings [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+def default_flavor():
+    return 'release'
+
+def get_common_config():
+    clang = 'clang++-5.0'
+    return {
+        'compiler': clang,
+        'linker': clang,
+        'compile_flags': ['-std=c++11', '-Wall', '-fcolor-diagnostics'],
```
Comment (dlehavi-resonai) [inline tests/cpp/YSettings]: we are currently on 14

File: yabt/builders/cpp.py [hunk 2]
```diff
@@ -42,6 +43,55 @@
 logger = make_logger(__name__)
 
 
+class CompilerConfig:
+    """Helper class for managing compiler / linker options and flags"""
+
+    def __init__(self, config, target):
+        self.compiler = self.get('compiler', config, target, 'g++')
```
Comment (dlehavi-resonai) [inline yabt/builders/cpp.py]: so, this means that the default is g++ and we are overiding it ?

File: yabt/builders/cpp.py [hunk 3]
```diff
@@ -145,32 +183,26 @@ def cpp_prog_builder(build_context, target):
     buildenv_workspace = build_context.conf.host_to_buildenv_path(
         workspace_dir)
     objects.extend(compile_cc(
-        build_context, target.props.in_buildenv, target.props.sources,
+        build_context, cc, target.props.in_buildenv, target.props.sources,
         workspace_dir, buildenv_workspace, target.props.cmd_env))
-    bin_file = join(buildenv_workspace, target.props.binary)
-    # TODO: linker flags should come from target & project conf
```
Comment (dlehavi-resonai) [inline yabt/builders/cpp.py]: why no TODO ? we have (important) linker flags

File: yabt/builders/cpp.py [hunk 4]
```diff
@@ -42,6 +43,55 @@
...
+# Common C++ builder signature terms
+CPP_SIG = [
+     ('sources', PT.FileList),
+     ('in_buildenv', PT.Target),
+     ('headers', PT.FileList, None),
+     ('cmd_env', None),
+     ('compiler', PT.str, None),
+     ('linker', PT.str, None),
+     ('compile_flags', PT.list, None),
+     ('link_flags', PT.list, None),
```
Comment (dlehavi-resonai) [inline yabt/builders/cpp.py]: where are these defaults set ?

File: tests/cpp/YSettings (hunk 1 above)
Comment (dlehavi-resonai) [reply tests/cpp/YSettings]: We are also using llvm libc and not gnus

File: yabt/builders/cpp.py (hunk 4 above)
Comment (itamaro) [reply yabt/builders/cpp.py]: at YBT level the default linker flags list is empty. at the project level, the defaults are set in YSettings, using the "settings API" that's demonstrated in the tests.

File: yabt/builders/cpp.py (hunk 3 above)
Comment (dlehavi-resonai) [reply yabt/builders/cpp.py]: but I don't see any linker flags !

File: yabt/builders/cpp.py (hunk 4 above)
Comment (dlehavi-resonai) [reply yabt/builders/cpp.py]: please add some custom linker flags (like you have with the compiler flags), for the lost souls reading this.

**Pull request merged**